pkg/api/validate/dynamic/dynamic.go

```diff
@@ -425,3 +452,42 @@ func uniqueSubnetSlice(slice []Subnet) []Subnet {
 	}
 	return list
 }
+
+func validateSkuLocation(ctx context.Context, location string, subscriptionID string, oc *api.OpenShiftCluster, skus []mgmtcompute.ResourceSku) error {
+	var (
+		workerProfileSku string
+		skuAvailable     bool
+	)
+
+	// Find desired worker SKU in cluster document's WorkerProfile
+	for _, workerprofile := range oc.Properties.WorkerProfiles {
+		if len(workerprofile.VMSize) != 0 {
+			workerProfileSku = string(workerprofile.VMSize)
+		} else {
+			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeNotFound, "", "Dynamic validation failed: Cluster WorkerProfile was not found.")
+		}
+	}
```

Comment:
What happens if there are multiple worker profiles?

Reply:
> Should we fail if there are multiple?

It is currently not possible to have multiple worker profiles during installation: we create only one which then is going to turn into either one (no AZ support) or three (az support) machine sets. Post-cluster installation customers can also add more machine sets with different VM sizes.

Try to `az aro create --debug` and see what goes into the RP as a post request. Then try `az aro show` to see what you get in cluster profiles. Reponce post-cluster creation gets enriched [here](https://github.com/Azure/ARO-RP/blob/0db1e3eb6e05a9b9bbaa2e1efae32d41ed83fde5/pkg/util/clusterdata/worker_profiles_task.go): machine sets get translated into worker profiles.

So during cluster creation - there will always be one profile. After - there will be multiple.

In the dynamic validator we should handle each worker profile & validate VM size for each profile. We call dynamic validator [on update](https://github.com/Azure/ARO-RP/blob/0db1e3eb6e05a9b9bbaa2e1efae32d41ed83fde5/pkg/cluster/install.go#L65). I think at the moment we do not call the enriching code which translates machinesets into worker profiles on update - we might need to do that at some point if, for example, we decide to allow modifications of worker profiles on update. Or allow multiple worker profiles at creation time. Or start using validator in more places (we already partly [use it in the operator](https://github.com/Azure/ARO-RP/blob/0db1e3eb6e05a9b9bbaa2e1efae32d41ed83fde5/pkg/operator/controllers/checker/serviceprincipalchecker.go#L82-L85) inside the clsuter).